dim/dim/rpc.py

```diff
@@ -1206,6 +1207,92 @@ def ippool_remove_vlan(self, pool):
+        self.user.can_network_admin()
+        pool = get_pool(pool)
+        from_layer3domain = pool.layer3domain
+        layer3domain = get_layer3domain(layer3domain)
+
+        pool.layer3domain = layer3domain
+        # check overlaps in target layer3domain and check for new parent
+        for subnet in pool.subnets:
+            parent = db.session.query(Ipblock) \
+                    .filter(Ipblock.layer3domain == layer3domain) \
+                    .filter(Ipblock.address <= subnet.address) \
+                    .filter(Ipblock.prefix < subnet.prefix) \
+                    .join(IpblockStatus).filter(IpblockStatus.name == 'Container') \
+                    .order_by(Ipblock.address.desc(), Ipblock.prefix) \
+                    .limit(1)
+            if parent.first() is None:
+                raise DimError('could not find new parent for subnet "%s" in new layer3domain' % (subnet.ip))
+
+            overlaps = db.session.query(Ipblock) \
+                    .filter(Ipblock.layer3domain == layer3domain) \
+                    .filter(Ipblock.address.between(subnet.address - 1, subnet.ip.broadcast.address)) \
+                    .filter(Ipblock.prefix >= subnet.prefix) \
+                    .count()
+            if overlaps > 0:
+                raise DimError('''subnet %s can't be moved, %d existing containers, subnets or static entries are in the way''' % (subnet.ip, overlaps))
+
+            overlaps = db.session.query(Ipblock) \
+                    .filter(Ipblock.layer3domain == layer3domain) \
+                    .filter(Ipblock.address <= subnet.address) \
+                    .filter(Ipblock.prefix < subnet.prefix) \
+                    .join(IpblockStatus).filter(IpblockStatus.name != 'Container') \
+                    .order_by(Ipblock.address.desc(), Ipblock.prefix) \
+                    .limit(1)
+            if (overlaps is not None and overlaps.first() is not None
+                    and subnet.parent is not None):
+                    overlap = overlaps.first()
+                    if (overlap.address >= subnet.address and overlap.address <= subnet.ip.broadcast.address) or \
+                       (overlap.ip.broadcast.address >= subnet.address and overlap.ip.broadcast.address <= subnet.ip.broadcast.address):
+                        raise DimError('''subnet %s can't be moved, %s already exists and is a %s''' % (subnet.ip, overlaps.first().ip, overlaps.first().status.name))
+
+            subnet.layer3domain = layer3domain
+            subnet.parent = parent.first()
+            Messages.info("Changing subnet %s to new parent %s in layer3domain %s" % (subnet.ip, parent.first().ip, layer3domain.name))
+
+            self._create_reverse_zones(subnet)
+            children = subnet.children.all()
+            for child in children:
+                logging.info("Changing child %s of subnet %s to layer3domain %s" % (child.ip, subnet.ip, layer3domain.name))
```

Comment:
That output is intentional. Because this hole operation is pretty slow and touches so much of the infrastructure, I want it to be logged, that these assets are being changed independently from the output and history.